mocha_test/every.js

```diff
@@ -83,6 +84,32 @@ describe("every", function () {
         });
     });
 
+    it('everySeries doesn\'t cause stack overflow (#1293)', function(done) {
+        var arr = _.range(10000);
+        let calls = 0;
+        async.everySeries(arr, function(data, cb) {
+            calls += 1;
+            async.setImmediate(_.partial(cb, null, false));
+        }, function(err) {
+            expect(err).to.equal(null);
+            expect(calls).to.equal(1);
+            done();
+        });
+    });
+
+    it('everyLimit doesn\'t cause stack overflow (#1293)', function(done) {
+        var arr = _.range(10000);
+        let calls = 0;
+        async.everyLimit(arr, 100, function(data, cb) {
+            calls += 1;
+            async.setImmediate(_.partial(cb, null, false));
```

Comment:
FYI, `setImmediate` will partially apply params for you, no need for `_.partial`